content/post/2023-06-06-system-dependencies/index.Rmd

```diff
@@ -0,0 +1,259 @@
+Depending on the case, the best course of action might be to:
+
+-   either edit the contents of `SystemRequirements` so that it's picked up by the regular expression
+-   or submit a pull request to `r-hub/r-system-requirements` if you believe the regular expression is too restrictive and should be updated
+
+Note however that the first option is likely always the simplest as it doesn't impact all the rest of the ecosystem (which is why `r-system-requirements` maintainers might be reluctant to relax a regular expression) and it is often something directly in your control, rather than a third-party.
+
+### Install system dependencies manually
+
+However, you might be in a case where the system dependency to install is not provided by package managers at all.
+Typically, if you had to compile or install it manually on your local computer, you're very likely to have to do the same operation in GitHub Actions.
+There two different, but somewhat equivalent, ways to do so, as detailed below.
+
+#### Directly in the GitHub Actions workflow
+
+You can insert the installation steps you used locally in the GitHub Actions workflow file.
+So, instead of having the usual structure, you have an extra step "Install extra system dependencies manually" that may look something like this:
+
+``` diff
+jobs:
+  R-CMD-check:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          use-public-rspm: true
+
++      - name: Install extra system dependencies manually
++        run:
++          wget ...
++          make
++          sudo make install
+
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          extra-packages: any::rcmdcheck
+          needs: check
+
+      - uses: r-lib/actions/check-r-package@v2
+```
+
+#### Using a Docker image in GitHub Actions
+
+Alternatively, you can do the manual installation in a Docker image and use this image in your GitHub Actions workflow.
+This is a particularly good solution if there is already a public Docker image or you already wrote a `DOCKERFILE` for your own local development purposes.
+If you use a public image, you can follow the steps in the official documentation to integrate it to your workflow.
+If you use a `DOCKERFILE`, you can follow [the answers to this stackoverflow question](https://stackoverflow.com/q/61154750/4439357) (in a nutshell, use `docker compose` in your job or publish the image first and then follow the official documentation).
+
+## Conclusion
+
+In this post, we have quickly reviewed how to specify system requirements for R package, how this seemingly innocent task requires a very complex infrastructure so that it can be understood by automated tools and that your dependencies are smoothly installed in a single command.
```

Comment:
```suggestion
In this post, we have provided an overview of how to specify system requirements for R package, how this seemingly innocent task requires a very complex infrastructure so that it can be understood by automated tools and that your dependencies are smoothly installed in a single command.
```